docs/IsolationNetworkPolicies.md

```diff
@@ -0,0 +1,210 @@
+
+    // Selects namespaces using cluster-scoped labels. This field follows standard label
+    // selector semantics; if present but empty, it selects all namespaces.
+    //
+    // If WorkloadPodSelector is also set, then the WorkloadLocation as a whole selects
+    // the workload pods matching WorkloadLocation in the namespaces selected by NamespaceSelector.
+    // Otherwise, it selects all Pods in the namespaces selected by NamespaceSelector.
+	// IF you want to select a specific namespace based on its name, it can be done by selecting an immutable label 
+	// `kubernetes.io/metadata.name` which is automatically set by the Kubernetes control plane. The value of the label 
+	// is the namespace name.
+    // +optional
+    NamespaceSelector *metav1.LabelSelector `json:"namespaceSelector,omitempty" protobuf:"bytes,2,opt,name=namespaceSelector"`
+
+    // IPBlock defines policy on a particular IPBlock. If this field is set then neither of the other fields can be.
+	// the structure of the IPBlock is defined at https://kubernetes.io/docs/reference/generated/kubernetes-api/v1.26/#ipblock-v1-networking-k8s-io
+    // +optional
+    IPBlock *IPBlock `json:"ipBlock,omitempty" protobuf:"bytes,3,rep,name=ipBlock"`
+} 
+```
+- A Plotter and relevant blueprints will provide the ingress/egress data for chain connected modules.
+
+- Deprecate the `workloadSelector` element, but continue to support it for backward compatibility. Process it as an 
+additional `workloadPodSelector` entry in the `WorkloadLocation` array, in a combination with namespace where a 
+`FybrikApplication` is deployed. For example, if a `FybrikApplication` is deployed in the `fybrik-notebook-sample` 
+namespace and a `workloadSelector` defines that a workload should be labeled as `app: my-notebook`, this will be equivalent to  
+```yaml
+workloadLocation:
+  workloadPodSelector:
+    matchLabels:
+      app: my-notebook
+  namespaceSelector:
+    matchLabels:
+      kubernetes.io/metadata.name: fybrik-notebook-sample
+```
+In other words, the following two `FybrikApplications` are equals
+```yaml
+apiVersion: app.fybrik.io/v1beta1
+kind: FybrikApplication
+metadata:
+  name: my-notebook
+  namespace: fybrik-notebook-sample
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadSelector:
+      matchLabels:
+        app: my-notebook
+...
+```
+```yaml
+apiVersion: app.fybrik.io/v1beta1
+kind: FybrikApplication
+metadata:
+  name: my-notebook
+  namespace: fybrik-notebook-sample
+  labels:
+    app: my-notebook
+spec:
+  selector:
+    workloadLocations:
+      - workloadPodSelector:
+          matchLabels:
+            app: my-notebook
+        namespaceSelector:
+          matchLabels:
+            kubernetes.io/metadata.name: fybrik-notebook-sample 
+...
+```
+- If a user wants to run his workloads in different location, he can define several `WorkloadLocation`, they all will be
```

Comment:
Why does he need to define different locations?  If he provides a label won't a k8s query get him all of the workloads with that label?  For multi-cluster, isn't it the role of the multi-cluster control plane to provide all of them across the workloads?  Not sure why we are asking the FybrikApplication user to provide this info, rather than relying on k8s to do it.

Reply:
>Why does he need to define different locations? 

A user(s) can deploy workloads in namespaces. And or out of K8s cluster.

> If he provides a label won't a k8s query get him all of the workloads with that label?

By default, K8s returns pods from the current namespace. NP's `PodSelector` works for the NP's instance namespace.

>For multi-cluster, isn't it the role of the multi-cluster control plane to provide all of them across the workloads? 

depends ...

> Not sure why we are asking the FybrikApplication user to provide this info, rather than relying on k8s to do it.

But even today we require from the user to define labels, what is the difference.